src/main/java/algorithms/curated170/easy/Permutations.java

```diff
@@ -0,0 +1,34 @@
+package algorithms.curated170.easy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Permutations {
+        public List<List<Integer>> permute(int[] nums) {
+            List<List<Integer>> output = new ArrayList<List<Integer>>();
+            backtrack(0, nums, output);
+            return output;
+        }
+
+        public void backtrack(int choiceIndex, int[] nums, List<List<Integer>> output) {
+            if (choiceIndex == nums.length) {
+                ArrayList<Integer> permutation = new ArrayList<Integer>();
+                for (int num : nums) {
+                    permutation.add(num);
+                }
```

Comment:
What are the approaches for converting `int[]` to `List<Integer>`?
Here are few other that I have found:
I don't think they can get any simpler than the current approach. It's good to know about them though.
`List<Integer> permutation = Arrays.stream(nums).boxed().collect(Collectors.toList());`
```
ArrayList<Integer> permutation = new ArrayList<Integer>();
Arrays.stream(nums).forEach(permutation::add);
```